src/ampJson.js

```diff
@@ -1,10 +1,19 @@
-export default function ampJson(data, component) {
+function sanitize(component) {
+  if (!component) {
+    return component;
+  }
+
+  return encodeURIComponent(component);
+}
+
+export default function ampJson(data, component, type) {
   if (!data || !component) return '';
+
   return Object.entries(data).map(([key, value]) => (
-    `<${component} id="${key}">
+    `<${sanitize(component)} id="${sanitize(key)}"${type ? ` type="${sanitize(type)}"` : ''}>
```

Comment:
I don't believe encodeURIComponent works on ID values, which can't have `%`.

I think you'll need a more manual process that prevents `</script>`, and `"` for attribute values, spaces and `>` for attribute names, and spaces and `>` for tag names.